mex/common/types/temporal_entity.py

```diff
@@ -0,0 +1,419 @@
+        try:
+            other = self.validate(other)
+        except TypeError:
+            raise NotImplementedError() from None
+        return bool(self.date_time > other.date_time)
+
+    def __str__(self) -> str:
+        """Render temporal entity with format fitting for its precision."""
+        return self.date_time.strftime(
+            TEMPORAL_ENTITY_FORMATS_BY_PRECISION[self.precision]
+        )
+
+    def __repr__(self) -> str:
+        """Render a presentation showing this is not just a datetime."""
+        return f'{self.__class__.__name__}("{self}")'
+
+
+class YearMonth(TemporalEntity):
+    """Partial date pattern that accepts yyyy or yyyy-MM format."""
+
+    def __init__(
+        self,
+        *args: Union[int, str, date, datetime, "TemporalEntity"],
+        tzinfo: Optional[tzinfo] = None,
+    ) -> None:
+        """Validate for precision (year or month) after initialization."""
+        super().__init__(*args, tzinfo=tzinfo)  # type: ignore
+        if self.precision not in (
+            TemporalEntityPrecision.YEAR,
+            TemporalEntityPrecision.MONTH,
+        ):
+            raise ValueError("Expected precision level 'YEAR' or 'MONTH'")
+
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls, _source: Type[Any], _handler: GetCoreSchemaHandler
+    ) -> core_schema.CoreSchema:
+        """Mutate the field schema for year-month pattern."""
+        from_str_schema = core_schema.chain_schema(
+            [
+                core_schema.str_schema(pattern=YEAR_MONTH_REGEX),
+                core_schema.no_info_plain_validator_function(
+                    cls.validate,
+                ),
+            ]
+        )
+
+        from_anything_schema = core_schema.chain_schema(
+            [
+                core_schema.no_info_plain_validator_function(cls.validate),
+                core_schema.is_instance_schema(YearMonth),
+            ]
+        )
+
+        return core_schema.json_or_python_schema(
+            json_schema=from_str_schema,
+            python_schema=from_anything_schema,
+        )
+
+    @classmethod
+    def __get_pydantic_json_schema__(
+        cls, core_schema_: CoreSchema, handler: GetJsonSchemaHandler
+    ) -> JsonSchemaValue:
+        """Modify the schema to add the class name as title and examples."""
+        json_schema = super().__get_pydantic_json_schema__(core_schema_, handler)
+        json_schema["examples"] = ["2011", "2019-03"]
+        return json_schema
+
+
+class YearMonthDay(TemporalEntity):
+    """Date pattern that accepts only yyyy-MM-dd format."""
+
+    def __init__(
+        self,
+        *args: Union[int, str, date, datetime, "YearMonthDay"],
+        tzinfo: Optional[tzinfo] = None,
+    ) -> None:
+        """Validate for day precision after initialization."""
+        super().__init__(*args, tzinfo=tzinfo)  # type: ignore
+        if self.precision != TemporalEntityPrecision.DAY:
```

Comment:
we could just put `ALLOWED_PRECISION_LEVELS` as a class var and then we could move this check to `TemporalEntity` and wouldn't have to have an `__init__` on the subclasses at all

Reply:
Another great idea, done in https://github.com/robert-koch-institut/mex-common/commit/1f6574f6ca9d6d9ae00852e44f51d79634ed3f5b